templates/types/streaming/nextjs/app/components/ui/chat/index.ts

```diff
@@ -57,8 +57,9 @@ export type EventData = {
 };
 
 export type AgentEventData = {
-  agent: string;
-  text: string;
+  name: string;
+  msg: string;
+  event_type: "text" | "progress";
```

Comment:
alternative format (TBD):
we're sending an event with an update of the status of a task (that sends its feedback as text)
text and that 
```suggestion
  agent: string;
  type: "task"; // can be extended with other types
  data: AgentTask
}
type AgentTask = {
  id: string; // (I guess you need this to group the progress)
  msg: string;
  progress?: ProgressData;
}
type ProgressData = {
  current: number;
  total: number;
}
```

Reply:
to keep changes minimal, how about we move `ProgressData` to here and make this change: 
```suggestion
  agent: string;
  data: string | ProgressData;
  type: "text" | "progress";
```

Reply:
Update, please check my comment: https://github.com/run-llama/create-llama/pull/403#discussion_r1821984741